### ISSUE TITLE
import numba crashed after import torchvision

### ISSUE THREAD

**Issue by KorcyES** [numba/numba#4870]
-   [x] I am using the latest released version of Numba (most recent is visible in
    the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
-   [x] I have included below a minimal working reproducer (if you are unsure how
    to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

If I import numba after torchvision:

```python
import torchvision
print("import torchvision done...")
import numba
print("import numba done...")
```

The program can only successfully import torchvision and seems to be stuck in a dead loop. There is nothing happened when I press Ctrl+C. When I use the 'top' command to check the system process, I found that the python process has disappeared.

Import in other way works fine:

```python
import numba
import torchvision
```

Versions:

Python 3.7.0 (default, Oct  9 2018, 10:31:47)
[GCC 7.3.0] :: Anaconda, Inc. on linux
Ubuntu 16.04.5 LTS (GNU/Linux 4.15.0-66-generic x86_64)
numba: 0.46
torchvision: 0.3.0

**Comment by seibert** [comment]
Do you see the same freeze if you import torch?:

```
 import torch
print("import torch done...")
import numba
print("import numba done...")
```

I wonder if this is an LLVM initialization conflict with PyTorch.

**Comment by KorcyES** [comment]
> Do you see the same freeze if you import torch?:
> 
> ```python
> import torch
> print("import torch done...")
> import numba
> print("import numba done...")
> ```
> 
> I wonder if this is an LLVM initialization conflict with PyTorch.

Yes，I see the same freeze after import torch. The program only output "import torch done..." and freezed

**Comment by stuartarchibald** [comment]
This seems to be fixed with the latest Numba (0.51.2) and PyTorch 1.5.0 :
```
$ conda list | grep 'torch \|numba '
numba                     0.51.2          np1.11py3.7h04863e7_g9d5709615_0    numba
pytorch                   1.5.0           cpu_py37hd91cbb3_0  

$ cat issue4870.py 
import torch
print("import torch done...")
import numba
print("import numba done...")

$ python issue4870.py 
import torch done...
import numba done...

```
Closing this issue, it is assumed that the above resolved the reported issue, if this is not the case please reopen the issue with new information. Many thanks.
